### ISSUE TITLE
[BUG] `npm unpublish <package>@<version> -f` unpublishes all versions

### ISSUE THREAD

**Issue by b4rtaz** [npm/cli#6786]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Hello!

Yesterday I published a new version of my package: `sequential-workflow-designer`. After the publication, I found a bug so I wanted to delete this version and publish the fixed version. I executed this command:

`npm unpublish sequential-workflow-designer@0.15.0 -f`

Unfortunately the version in the command has not been taken into account, and npm unpublished the whole package.

### Expected Behavior

The cli should unpublish only the specific version.

### Steps To Reproduce

```
npm unpublish <package>@<version> -f
```

### Environment

- npm: 9.6.7
- Node.js: v18.17.0
- OS Name: macOS Ventura 13.5 (22G74)

**Comment by ljharb** [comment]
Why did you use -f before trying it without that option? -f is “force” and is an extremely dangerous modifier.

**Comment by b4rtaz** [comment]
Yes, I know. I couldn't unpublish `sequential-workflow-designer` without the `-f` modifier because there was a dependent package that used 0.15.0 (`sequential-workflow-designer-react`). Yes I know, I could start from the dependent package, or just release a next version.

But still, I just wanted to unpublish a specific version, not the whole package.

**Comment by ljharb** [comment]
If you can’t unpublish a version for that reason, then you can’t unpublish it with or without -f. npm’s logic doesn’t care which version is important, afaik, it blocks unpublishing of every version of any version is unpublishable.

I’m definitely surprised that an unpublish command would be able to remove more than one version at a time, to be clear.

**Comment by b4rtaz** [comment]
> If you can’t unpublish a version for that reason, then you can’t unpublish it with or without -f. npm’s logic doesn’t care which version is important, afaik, it blocks unpublishing of every version of any version is unpublishable.

My package is used by 5+ packages. So if it would work as you described then I couldn't unpublish any version. In fact, by single command I unpublished all versions.
